test/parallel/test-trace-events-process-exit.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+const fs = require('fs');
+
+const FILE_NAME = 'node_trace.1.log';
+
+common.refreshTmpDir();
+process.chdir(common.tmpDir);
+
+const proc = cp.spawn(process.execPath,
+                      [ '--trace-events-enabled',
+                        '-e', 'process.exit()' ]);
+
+proc.once('exit', common.mustCall(() => {
+  assert(common.fileExists(FILE_NAME));
+  fs.readFile(FILE_NAME, common.mustCall((err, data) => {
```

Comment:
Use `fs.readFileSync` here, the async version is not guaranteed to complete.  In fact, it probably doesn't but that goes unnoticed because the `common.mustCall(...)` isn't checked as you're already inside an exit handler.

Reply:
This is a child process `'exit'` event, not `process.exit()`. `fs.readFileSync()` is probably a simpler idea either way though.

Reply:
This is how all of the other `trace-events` tests work. I can change it, but at least it should be consistent.